sklearn/ensemble/tests/test_voting.py

```diff
@@ -674,3 +680,97 @@ def test_get_features_names_out_classifier_error():
     )
     with pytest.raises(ValueError, match=msg):
         voting.get_feature_names_out()
+
+
+# Metadata Routing Tests
+# ======================
+
+
+@pytest.mark.parametrize("Estimator", [VotingClassifier, VotingRegressor])
+def test_routing_passed_metadata_not_supported(Estimator):
```

Comment:
```suggestion
@pytest.mark.parametrize(
    "Estimator, Child",
    [(VotingClassifier, ConsumingClsasifier), (VotingRegressor, ConsumingRegressor)],
)
def test_routing_passed_metadata_not_supported(Estimator, Child):
```

You can pass the child here instead of the if else condition bellow.